Client/Frontend/Browser/Tabs/TabManagerMiddleware.swift

```diff
@@ -161,8 +161,21 @@ class TabManagerMiddleware {
     }
 
     private func selectTab(for tabUUID: String) {
-        guard let tab = tabManager.getTabForUUID(uuid: tabUUID) else { return }
+        guard let tab = defaultTabManager.getTabForUUID(uuid: tabUUID) else { return }
+
+        defaultTabManager.selectTab(tab)
+    }
+
+    private func setTabManager(_ tabManager: TabManager, for sceneUUID: SceneUUID) {
+        tabManagers[sceneUUID] = tabManager
+    }
+
+    private func removeTabManager(_ tabManager: TabManager, for sceneUUID: SceneUUID) {
+        tabManagers.removeValue(forKey: sceneUUID)
+    }
 
-        tabManager.selectTab(tab)
+    private var defaultTabManager: TabManager {
+        // TODO: [7863] Temporary. WIP for Redux + iPad Multi-window.
+        return tabManagers[WindowData.DefaultSingleWindowUUID]!
```

Comment:
@OrlaM @yoanarios One thing I wasn't a fan of is the IUO here. I had looked at a few different ways of avoiding this, but the best I could find was to simply make this a proper optional and return `TabManager?`. That would mean adding some extra guards/checks for nil in the related `tabsPanelProvider()` code.

I'd prefer to go that route, but I wanted to be check if anyone had concerns? I didn't want to change too much of Yoana's nice new code here without discussing first.

Reply:
This is the equivalent behaviour to if we pulled this from the dependency container, the failure response is to crash. I don't love it, but in the absence of a good alternative I'm fine with it.